siriuspy/siriuspy/diagsys/lidiag/pvs.py

```diff
@@ -134,6 +134,178 @@ def compute_update(self, computed_pv, updated_pv_name, value):
+    BIT_TRGNORM = 0b000100000000000000
+    BIT_PLSCURR = 0b001000000000000000
+    BIT_VLTDIFF = 0b010000000000000000
+    BIT_CRRDIFF = 0b100000000000000000
+
+    PV_RUNSTOP = 0
+    PV_PREHEAT = 1
+    PV_CHRALLW = 2
+    PV_TRGALLW = 3
+    PV_EMRSTOP = 4
+    PV_CPS_ALL = 5
+    PV_THYHEAT = 6
+    PV_KLYHEAT = 7
+    PV_LVRDYOK = 8
+    PV_HVRDYOK = 9
+    PV_TRRDYOK = 10
+    PV_SELFFLT = 11
+    PV_SYS_RDY = 12
+    PV_TRGNORM = 13
+    PV_PLSCURR = 14
+    PV_VLTDIFF = 15
+    PV_CRRDIFF = 16
+
+    def compute_update(self, computed_pv, updated_pv_name, value):
+        """Compute Status PV."""
+        value = 0
+
+        # connected?
+        disconnected = \
+            not computed_pv.pvs[LIPUStatusPV.PV_RUNSTOP].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_PREHEAT].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_CHRALLW].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_TRGALLW].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_EMRSTOP].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_CPS_ALL].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_THYHEAT].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_KLYHEAT].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_LVRDYOK].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_HVRDYOK].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_TRRDYOK].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_SELFFLT].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_SYS_RDY].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_TRGNORM].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_PLSCURR].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_VLTDIFF].connected or \
+            not computed_pv.pvs[LIPUStatusPV.PV_CRRDIFF].connected
+        if disconnected:
+            value |= LIPUStatusPV.BIT_DISCONN
+            value |= LIPUStatusPV.BIT_RUNSTOP
+            value |= LIPUStatusPV.BIT_PREHEAT
+            value |= LIPUStatusPV.BIT_CHRALLW
+            value |= LIPUStatusPV.BIT_TRGALLW
+            value |= LIPUStatusPV.BIT_EMRSTOP
+            value |= LIPUStatusPV.BIT_CPS_ALL
+            value |= LIPUStatusPV.BIT_THYHEAT
+            value |= LIPUStatusPV.BIT_KLYHEAT
+            value |= LIPUStatusPV.BIT_LVRDYOK
+            value |= LIPUStatusPV.BIT_HVRDYOK
+            value |= LIPUStatusPV.BIT_TRRDYOK
+            value |= LIPUStatusPV.BIT_SELFFLT
+            value |= LIPUStatusPV.BIT_SYS_RDY
+            value |= LIPUStatusPV.BIT_TRGNORM
+            value |= LIPUStatusPV.BIT_PLSCURR
+            value |= LIPUStatusPV.BIT_VLTDIFF
+            value |= LIPUStatusPV.BIT_CRRDIFF
+            return {'value': value}
+
+        # Run/Stop
+        sts = computed_pv.pvs[LIPUStatusPV.PV_RUNSTOP].value
+        if sts != 1 or sts is None:
+            value |= LIPUStatusPV.BIT_RUNSTOP
```

Comment:
if the BIT constant were the number of the bit and not the value, you would be able to use our method `update_bit` like this:
`update_bit(value, LIPUStatusPV.BIT_RUNSTOP, <sts_logic>)`

Reply:
done!